test/common/http/conn_manager_impl_test.cc

```diff
@@ -2452,13 +2459,16 @@ TEST_F(HttpConnectionManagerImplTest, TestPeriodicAccessLogging) {
                      const HeaderMap*, const StreamInfo::StreamInfo& stream_info) {
             EXPECT_EQ(&decoder_->streamInfo(), &stream_info);
             EXPECT_THAT(request_headers, testing::NotNull());
-            EXPECT_THAT(response_headers, testing::IsNull());
-            EXPECT_EQ(stream_info.requestComplete(), absl::nullopt);
+            EXPECT_THAT(response_headers, testing::NotNull());
+            EXPECT_EQ(StreamInfo::StreamState::InProgress, stream_info.streamState());
           }));
+
   // Pretend like some 30s has passed, and the log should be written.
   EXPECT_CALL(*periodic_log_timer, enableTimer(*access_log_flush_interval_, _)).Times(2);
+
   periodic_log_timer->invokeCallback();
```

Comment:
add check for invoking periodic callback before the streamState() == InProgress. i.e.

Remove encodeHeaders call from the `dispatch` callback above.
add a EXPECT_CALL(*handler, log(..)).Times(0) after the `onData` call
then call `encodeHeaders`.